homeassistant/components/number/significant_change.py

```diff
@@ -0,0 +1,94 @@
+"""Helper to test significant Number state changes."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_UNIT_OF_MEASUREMENT,
+    PERCENTAGE,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.significant_change import (
+    check_absolute_change,
+    check_percentage_change,
+    check_valid_float,
+)
+
+from .const import NumberDeviceClass
+
+
+def _absolute_and_relative_change(
+    old_state: int | float | None,
+    new_state: int | float | None,
+    absolute_change: int | float,
+    percentage_change: int | float,
+) -> bool:
+    return check_absolute_change(
+        old_state, new_state, absolute_change
+    ) and check_percentage_change(old_state, new_state, percentage_change)
+
+
+@callback
+def async_check_significant_change(
+    hass: HomeAssistant,
+    old_state: str,
+    old_attrs: dict,
+    new_state: str,
+    new_attrs: dict,
+    **kwargs: Any,
+) -> bool | None:
+    """Test if state significantly changed."""
+    absolute_change: float | None = None
+    percentage_change: float | None = None
+
+    # default for non classified
+    if (device_class := new_attrs.get(ATTR_DEVICE_CLASS)) is None:
+        absolute_change = 1.0
+        percentage_change = 2.0
```

Comment:
I think that is it isn't classified, it is hard to judge if the change is significant or not?

Reply:
maybe we should return `None` in that case?

Reply:
Shouldn't we consider it significant in those cases for any change? (as we don't known)

Reply:
In the sensor platform, we return None in case it is unclassified ... tbh this sounds reasonable, because we can not judge it's significance and as per docs `None` means "we so not know" 🤔

Reply:
Sounds good 👍